provisioning/miniprov/hare_mp/main.py

```diff
@@ -144,6 +144,15 @@ def get_server_type(url: str) -> str:
         logging.error('Cannot get server type (%s)', error)
         return 'unknown'
 
+def is_stage_required(url: str) -> str:
+    try:
+        provider = ConfStoreProvider(url)
+        machine_id = provider.get_machine_id()
+        pod_type = provider.get(f'node>{machine_id}>type')
+        return pod_type == 'server_node'
```

Comment:
It is now decided that pod_type or node_type is a dynamic key and we need to use constants from https://github.com/Seagate/cortx-utils/pull/640/files.
We can use `SERVICE_MOTR_IO` macro to search for io services, thus, if ioservices are configured on a node then execute mkfs else not. We need to use conf-store's search api as below,
e.g.
```
Conf.search(self.default_index, parent_key, key_to_search, value_to_search)
```

Reply:
@mssawant Please look at my comment regarding this call [here](https://github.com/Seagate/cortx-hare/pull/1904#discussion_r769388348).

Summary: I suggest not using this API call. It doesn't help us much anyway (so @rahul27kumar has to filter out the resulting keys manually afterwards - and that's said to be "expected").

Reply:
@knekrasov, Problem is, as per the provisioning team, there are some keys that are volatile and some constant. Provisioning team has introduced a list of constants that we can use, so atleast we have a contract in some form.
So, it is suggested to search and extract the information from conf-store rather than accessing a keys directly.